services/secondary-storage/server/src/org/apache/cloudstack/storage/resource/NfsSecondaryStorageResource.java

```diff
@@ -2809,7 +2808,7 @@ private synchronized void checkSecondaryStorageResourceLimit(TemplateOrVolumePos
         int accountDirSizeInGB = getSizeInGB(accountTemplateDirSize + accountSnapshotDirSize + accountVolumeDirSize);
         int defaultMaxAccountSecondaryStorageInGB = Integer.parseInt(cmd.getDefaultMaxAccountSecondaryStorage());
 
-        if ((accountDirSizeInGB + contentLengthInGB) > defaultMaxAccountSecondaryStorageInGB) {
+        if (defaultMaxAccountSecondaryStorageInGB != Resource.RESOURCE_UNLIMITED && (accountDirSizeInGB + contentLengthInGB) > defaultMaxAccountSecondaryStorageInGB) {
```

Comment:
@niteshsarda thanks for the fix.
To improve this PR to an even higher level, how do you feel about extracting the condition of the `if` to a method? 

Then, this method can receive test cases and Java doc explaining when it returns true or false. 
This can facilitate code maintenance in the future and also to reduce the bar for newcomers.

Reply:
@rafaelweingartner : In my opinion, we don not need a separate method for this code because both the conditions in "if " loop are pretty simple to understand.
Also, we are using Java doc very rarely in our code, so adding Java doc for this straight forward condition will not be much useful.

Reply:
I agree with @niteshsarda

Reply:
No problem, but I disagree with this kind of thinking "we are using Java doc very rarely in our code, so adding Java doc for this straight forward condition will not be much useful".

This is what makes our code base terrible in same (most) places and in my opinion we should improve bit by bit. Code  (easy to maintain, easy to be extended by newcomers, and easy to read) is a culture that we have to create.

Reply:
I do agree with @rafaelweingartner to some extend but I don't think we should stop this fix because of it. Also the extend is that I agree that extracting the if in a separate method but that we should always try to not need any comments (including javadoc)
I am merging this.

Reply:
@DaanHoogland agreed with you ;)

Sometimes people do not understand when I suggest improvements. In this case for example, my intention was not to stop the addition of the code, but rather to improve it, and to show this perspective to the contributor. If I am against a code being merged, I will for sure tell that directly.

Thanks for merging.